### PR TITLE
Feature: Lecture Modularisation and Learning Goals | Database Changes

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
### Checklist
- [X] Server: I documented the Java code using JavaDoc style.

### Motivation and Context

This PR contains only the entity/database changes necessary for the lecture modularisation and learning goals future described in: https://github.com/ls1intum/Artemis/issues/2205

#### Class Diagram
image

### Description

Creation of entities, repositories and liquibase changelog.

### Steps for Testing
Look through the code and check that the necessary relationships are implemented there

### Test Coverage
No logic in this PR, thus nothing to test.

### CODE REVIEW

**Comment by krusche** [comment]
Module -> Unit

**Comment by stefanwaldhauser** [comment]
> Module -> Unit

done

File: src/main/java/de/tum/in/www1/artemis/domain/Exercise.java [hunk 1]
```diff
@@ -523,7 +545,7 @@ else if (participation.getExercise() instanceof ModelingExercise || participatio
      * Get the latest relevant result from the given participation (rated == true or rated == null) (relevancy depends on Exercise type => this should be overridden by subclasses
      * if necessary)
      *
-     * @param participation the participation whose results we are considering
+     * @param participation           the participation whose results we are considering
```
Comment (FrankeLukas) [inline src/main/java/de/tum/in/www1/artemis/domain/Exercise.java:548]: This change doesn't look correct.

Comment (stefanwaldhauser) [reply src/main/java/de/tum/in/www1/artemis/domain/Exercise.java:548]: Thats how intellj formats the file automatically.

**Comment by kloessst** [comment]
Can u update feature/lecture-modules-learning-goals/database by merging develop into it? Otherwise the diff is cluttered with changes unrelated to your PR.

**Comment by stefanwaldhauser** [comment]
DB Changed will be included in https://github.com/ls1intum/Artemis/pull/2218

**Pull request closed without merging**